src/ducks/Signals/signalModal/SignalMasterModalForm.js

```diff
@@ -49,32 +55,60 @@ const SignalMasterModalForm = ({
   const { variant, border } = buttonParams
 
   return (
-    <Dialog
-      open={dialogOpenState}
-      onOpen={() => {
-        setDialogOpenState(true)
+    <GetSignal
+      triggerId={triggerId}
+      render={({ trigger = {} }) => {
+        const { isLoading, isError } = trigger
+
+        if (isShared && trigger.trigger) {
+          trigger.trigger = { ...trigger.trigger, ...shareParams }
+        }
+
+        if (isError) {
+          throw new Error(`Can't find such public trigger with id ${triggerId}`)
+        }
+
+        return (
+          <Dialog
+            open={dialogOpenState}
+            onOpen={() => {
+              setDialogOpenState(true)
+            }}
+            onClose={onClose}
+            trigger={signalModalTrigger(
+              isLoggedIn && enabled,
+              label,
+              variant,
+              border
+            )}
+            title={
+              !isError && (
+                <>
+                  {dialogTitle}
+                  {isShared && <div className={styles.shared}>Shared</div>}
+                </>
+              )
+            }
+            classes={styles}
+          >
+            <Dialog.ScrollContent className={styles.TriggerPanel}>
+              {isLoading && <div className={styles.loading}>Loading...</div>}
+              {!isLoading && (
```

Comment:
Can we use animation component for Loading state? `Loader` from `sun-ui` for ex.

Reply:
Replaced